hera_sim/rfi.py

```diff
@@ -264,14 +264,17 @@ def rfi_dtv(fqs, lsts, rfi=None, freq_min=.174, freq_max=.214, width=0.008,
         raise ValueError("strength_std must be float or list with len equal to number of bands")
 
     for band, chnc, strngth, str_std in zip(bands, chance, strength, strength_std):
-        fq_ind_min = np.argwhere(band <= fqs)[0][0]
-        fq_ind_max = fq_ind_min + int(width / delta_f) + 1
-        this_rfi = rfi[:, fq_ind_min:min(fq_ind_max, fqs.size)]
-
-        rfis = np.random.uniform(size=lsts.size) <= chnc
-        this_rfi[rfis] += np.atleast_2d(np.random.normal(strngth, str_std, size=np.sum(rfis)) * np.exp(
-            2 * np.pi * 1j * np.random.uniform(size=np.sum(rfis))
-        )).T
+        try:
+            fq_ind_min = np.argwhere(band <= fqs)[0][0]
+            fq_ind_max = fq_ind_min + int(width / delta_f) + 1
+            this_rfi = rfi[:, fq_ind_min:min(fq_ind_max, fqs.size)]
+
+            rfis = np.random.uniform(size=lsts.size) <= chnc
+            this_rfi[rfis] += np.atleast_2d(np.random.normal(strngth, str_std, size=np.sum(rfis)) * np.exp(
+                2 * np.pi * 1j * np.random.uniform(size=np.sum(rfis))
+            )).T
+        except IndexError:
+            pass
```

Comment:
Is it reasonable to always just pass silently here? Shouldn't we be filling in the part of the band that is DTV if it overlaps with an edge?

Reply:
That's a good point. I figure what we could do, then is modify the line `fq_ind_min = np.argwhere(band <= fqs)[0][0]` to instead be `fq_ind_min = np.argwhere(band + width <= fqs)[0][0]`, but keep the `try: except:` clause in there to catch the case where `np.argwhere` returns an empty list. Does that make sense?

Reply:
I actually think that might have some unwanted behavior... I'll have to think about this a little more carefully.

Reply:
After reading the documentation and thinking about it, I think that it *is* reasonable to pass silently here. The `IndexError` is raised only in the case that `np.all(fqs < band)` is `True`. Since `band` is the lower end of the DTV channel, if an `IndexError` is raised, then all of the frequencies in that DTV channel are strictly greater than all of the frequencies passed to `rfi_dtv`.

I think that the developer of this code figured that the following clause would ensure that this never happens:
```
if fqs.max() <= bands.max():
    bands = bands[:-1]
```
This works fine under the assumption that there is at least one frequency in `fqs` that is greater than `bands.min()`, but it breaks for the case that `fqs.max() < bands.min()` is `True`.

Reply:
A quick follow-up to clarify slightly: if `fqs.max() < bands.min()`, then `np.argwhere(band <= fqs)[0][0]` will raise an `IndexError` for every `band` in `bands`; however, if `fqs.max() > bands.min()`, then the line `if fqs.max() <= bands.max(): bands = bands[:-1]` should ensure that `np.argwhere(band <= fqs)` will never return an empty array.

Reply:
LOL, I was the developer of this code... I can't remember what I did or why I did it!

Having a look now, you probably just want to do something like:

```
bands = bands[np.logical_and(bands >= fqs.min(), bands < fqs.max())]
```

just after creating the `bands` array. This will ensure all bands are within the freq range.

Reply:
Ok, I agree that this will ensure that all bands are within the freq range; however, if the simulated frequencies are all below the lower-edge of the DTV band (which is the case for the `Simulator` object made from `create_sim` in the `test_simulator` module), then this will return an empty array, which will cause `np.argwhere(band <= fqs)[0][0]` to throw an `IndexError`.

Reply:
So I think it makes sense to keep the `try: except:` clause, but I think it may be a good idea to add a warning notifying the user that their simulated frequencies are outside of the DTV channels.

Reply:
I don't think this will throw an error: if `bands` is empty, then the loop won't be entered at all.

One case that's not quite covered by my suggestion is where `freq_min` is smaller than `fqs.min()`, but only just, so that part of that DTV band overlaps with the lowest channels. With my suggestion, this band gets cut entirely. You'll probably have to special-case that. 

I'm up for still using the try: except: anyway, in case we botch something. Or perhaps it's better to use try: except: than limiting the `bands` array: scroll through all bands, and if its close enough or within `fqs`, do what is necessary, and otherwise just warn the user -- it's likely that if this happens they have done something wrong themselves.

In this case, you have to special case where `band` is less than `fqs.min() - width`, and warn the user. Also special-case where `band` is between `fqs.min() - width` and `fqs.min()`, and use the first few indices.

Reply:
Hmm. Should I just raise an issue on this and work on figuring it out there? This *is* a separate problem from the actual thing I'm trying to get pulled in here.

Reply:
Yes, that would be great. However, the tests have to all work for it to be merged.

Reply:
So it seems like there are two ways for me to make sure the tests will work. The first is to just not try to add RFI from DTV when testing `run_sim`. The second is to leave in the `try: except:` clause, and then either include or exclude DTV from the `run_sim` test. Either way, `rfi_dtv` can't really be tested in the `test_simulator` module, since the `create_sim` function initializes a `Simulator` object whose frequencies range from 100 MHz to (100 + 1000/1024) MHz, which is well below the DTV range.

Reply:
Yup, just leave it out of the tests for now, I reckon

Reply:
Cool. All the tests are passing. Can I finally get this thing merged in?